File: fetch_exchange.rb
# frozen_string_literal: true

require 'uri'
require 'net/http'
require 'json'

def fetch_exchange_rate(base, symbol)
  url = URI('https://data.fixer.io/api/latest')
  params = {
    access_key: '',
    base: base,
    symbols: symbol
  }
  url.query = URI.encode_www_form(params)
  response = Net::HTTP.get_response(url)

  raise "HTTP request failed: #{response.code}" unless response.is_a?(Net::HTTPSuccess)

  parsed_json = JSON.parse(response.body)

  raise "API request failed: #{parsed_json['error']['info']}" unless parsed_json['success']

  puts parsed_json['rates'][symbol]
rescue JSON::ParserError => e
  puts "Failed to parse JSON: #{e.message}"
end

fetch_exchange_rate('USD', 'ZAR')

File: fetch_exchange.c
#include <stdio.h>
#include <curl/curl.h>

#define API_KEY ""
#define BASE "USD"
#define SYMBOLS "ZAR"

int main(void) {
	const char* url = "http://data.fixer.io/api/latest?access_key="
					 API_KEY
					 "&base="
					 BASE
					 "&symbols="
					 SYMBOLS;

	CURL *curl;
	CURLcode res;

	curl_global_init(CURL_GLOBAL_DEFAULT);

	curl = curl_easy_init();
	if(curl) {
		curl_easy_setopt(curl, CURLOPT_URL, url);
		res = curl_easy_perform(curl);
		if(res != CURLE_OK) {
		    fprintf(stderr, "curl_easy_perform() failed: %s\n", curl_easy_strerror(res));
		}
		curl_easy_cleanup(curl);
	}
	
	curl_global_cleanup();
	   
	return 0;
}

